### PR TITLE
SAK-42577: reset-pass > no longer checks invalid account types

### PR DESCRIPTION
**Pull request by bjones86**
https://jira.sakaiproject.org/browse/SAK-42577

SAK-42256 refactored reset-pass so that it would not leak information to the user about whether or not an account exists with the given email address.

However, we didn't notice that as a result of this work there is now a regression in that it doesn't report/validate on a sakai.property to reject attempts for certain account types. Unfortunately, we can't just restore this original functionality because that would be defeating the purpose of hardening the system to not expose undue information to a potential attacker.

To address this regression, we actually need to change the original implementation to be based off a sakai.property like invalidEmailInIdAccountString, which lists domains which are not allowed to use services like New Account, rather than a list of allowed user types (which forces us to resolve User objects in the back end to identify the user type, and reporting on this finding divulges information about if the account exists or not).

In this way, the domain can be checked before trying to resolve the User object, we can display some messaging about the domain being invalid or not allowed, and we don't leak any information about if the account exists or not (either via messaging, or by code analysis).

This is a significant change in behvaiour, and will necessitate removing old sakai.properties ("resetPass.resetAllRoles", "accountValidator.accountTypes.accept", and "resetRoles" which is the legacy version of the former), and introducing a new property to duplicate the "invalidEmailInIdAccountString" for Reset Password, so that institutions can define two separate lists of allowed domains, one for New Account and one for Reset Password.

### CODE REVIEW

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 1]
```diff
@@ -3838,21 +3838,15 @@
...
-# resetPass.resetAllRoles=true
-
 # If set to false then password reset users get sent a new email, otherwise they get a link to allow
 # them to reset their password. This prevents people from changing password they don't own.
 # DEFAULT: false
 # siteManage.validateNewUsers=true
 
+# Comma-separated list of domain names that are not allowed to use the Reset Password system
+# DEFAULT: empty/null (all domains are allowed to use the service)
+# resetPass.invalidEmailDomains=sakaiproject.org
```
Comment (mpellicer) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties]: Really minor, sakailms.org instead.

File: reset-pass/reset-pass/src/java/org/sakaiproject/tool/resetpass/UserValidator.java [hunk 2]
```diff
@@ -53,65 +70,60 @@ public void setSecurityService(SecurityService ss){
 		this.securityService = ss;
 	}
 
+	private ToolManager toolManager;
```
Comment (mpellicer) [inline reset-pass/reset-pass/src/java/org/sakaiproject/tool/resetpass/UserValidator.java]: Codacy complaints about this, I've approved it anyway.

**Comment by bjones86** [comment]
Thanks, @mpellicer. I've updated the property to `sakailms.org`

**Comment by mpellicer** [comment]
Thanks @bjones86

**Pull request merged**